server/application/application.go

```diff
@@ -1399,9 +1399,14 @@ func (s *Server) WatchResourceTree(q *application.ResourcesQuery, ws application
 		return err
 	}
 
-	return s.cache.OnAppResourcesTreeChanged(ws.Context(), q.GetApplicationName(), func() error {
+	cacheKey := q.GetApplicationName()
+	if ns := q.GetAppNamespace(); ns != "" && ns != s.ns {
+		cacheKey = fmt.Sprintf("%s_%s", ns, cacheKey)
+	}
```

Comment:
Please use `AppInstanceName` from `util/argo` to get the cache key instead of rolling your own.

Reply:
Didn't notice we had this. Thanks!